src/block/Sign.php

```diff
@@ -142,6 +142,14 @@ public function getText() : SignText{
 		return $this->text;
 	}
 
+	/**
+	 * @param SignText $text
+	 */
+	public function setText(SignText $text) : void{
+		$this->text = $text;
+		$this->pos->getWorld()->setBlock($this->pos, $this);
```

Comment:
please remove this, it's possible that we may want to set the text of a block which doesn't have a position.